### DISCUSSION TITLE
Destroying a sprite from a container move other sprite

### DISCUSSION THREAD

**Discussion by KreNtal** [pixijs/pixijs#9059] [Q&A] [answered]
So here's a little demo to show the problem:

[https://codepen.io/KreNtal/pen/VwBbVjV](https://codepen.io/KreNtal/pen/VwBbVjV)

Basically when the last sprite is destroyed the new last sprite inside the container moves some pixels back.

Why is this happening? How can I avoid it?
Some clues will be really much appreciated, thanks.

**Comment by KreNtal** [comment, marked as answer]
So I'm just dumb, all I had to do was adding newTruck on index 0 with:

`addChildAt(newTruck, 0);`

Sorry for wasting your time.

**Reply by bigtimebuddy** [reply]
No problem, we are all still learning!

One thing I noticed is that you are destroying and creating new Sprites within the main ticker loop. This is usually a tell-tale sign of a potential memory leak or at the very least, results in hiccups in the JS garabage collector. It's better to [pool objects](https://medium.com/vngrs/javascript-object-pooling-47d888e1e2bf) into an array and pop/push them when you need instead of create/destroy.

**Reply by KreNtal** [reply]
Thanks for the hint, actually in the real application I do use object pooling. That was just a quick demo to show the problem.
But now that you pointed it out I will re-check it just to be sure. Thanks a bunch.
